tests/tests_unit/rules/test_analysis.py

```diff
@@ -71,4 +71,15 @@ def test_inherited_referred_classes(self, olav_rules: InformationRules) -> None:
         assert len(InformationArchitectRulesAnalysis(olav_rules).inherited_referred_classes) == 1
 
     def test_referred_classes(self, olav_rules: InformationRules) -> None:
-        assert len(InformationArchitectRulesAnalysis(olav_rules).referred_classes) == 4
+        expected = {
+            ClassEntity(prefix="power", suffix="GeoLocation"),
+            ClassEntity(prefix="power", suffix="Point"),
+            ClassEntity(prefix="power", suffix="Polygon"),
+        }
+
+        actual = InformationArchitectRulesAnalysis(olav_rules).referred_classes
+
+        missing = expected - actual
+        assert not missing, f"Missing classes: {missing}"
+        extra = actual - expected
+        assert not extra, f"Extra classes: {extra}"
```

Comment:
I changed this test. It used to be 4 due to the fact that `ParentClassEntity` and `ClassEntity` used to be considered not the same entity:
![image](https://github.com/cognitedata/neat/assets/60234212/c64df3c3-d7b0-4aaa-a5fb-62f858deab2d)

Now, since `ParentClassEntity` is inheriting from `ClassEntity` they are considered equal, so `ParentClassEntity(prefix="power", suffix="GeoLocation")` is removed.